draft-ietf-tls-dtls-connection-id.md

```diff
@@ -310,33 +313,51 @@ that do not use the with Encrypt-then-MAC processing
 described in {{RFC7366}}.
 
 ~~~
-    MAC(MAC_write_key, seq_num +
+    MAC(MAC_write_key,
+        seq_num_placeholder +
+        tls12_cid +
+        cid_length +
         tls12_cid +
         DTLSCiphertext.version +
+        epoch +
+        sequence_number +
         cid +
-        cid_length +
         length_of_DTLSInnerPlaintext +
         DTLSInnerPlaintext.content +
         DTLSInnerPlaintext.real_type +
         DTLSInnerPlaintext.zeros
-    )
+    );
 ~~~
 
+The rationale behind this construction is to separate the MAC input
+for DTLS without the connection ID from the MAC input with the
+connection ID. The former always consists of a sequence number
+followed by some other content type than tls12_cid; the latter
+always consists of the seq_num_placeholder followed by tls12_cid.
+Although 2^64-1 is potentially a valid sequence number, tls12_cid
+will never be a valid CID when the connection ID is not in use.
+In addition, the epoch and sequence_number are now fed into
```

Comment:
> will never be a valid CID when the connection ID is not in use.

I think, this should be:

> tls12_cid will never be one of the other content types when the connection ID is not in use.

The other MAC (when connection ID is not in use) uses the MAC of RFC5246 

```
MAC(MAC_write_key, seq_num +
                            TLSCompressed.type +
                            TLSCompressed.version +
                            TLSCompressed.length +
                            TLSCompressed.fragment);
```

not the one previously defined for this extension.

Reply:
I'm not sure I follow the distinction you are trying to draw.

Reply:
I think we're looking for s/never be a valid CID/never be a valid content type/

Reply:
> I think we're looking for s/never be a valid CID/never be a valid content type/

Yes.

Reply:
Thanks. Fixed.